aim2dat/strct/strct_manipulation.py

```diff
@@ -75,14 +76,8 @@ def substitute_elements(
     attributes2keep = ["space_group", "source"]
     str_el_pairs = None
     if any(el_pair[0] in structure["elements"] for el_pair in elements):
-        new_structure = {
-            "label": structure.label,
-            "elements": list(structure["elements"]),
-            "positions": structure["scaled_positions"],
-            "is_cartesian": False,
-            "pbc": structure["pbc"],
-            "attributes": {},
-        }
+        new_structure = structure.to_dict(cartesian=False)
+        new_structure["elements"] = list(new_structure["elements"])
```

Comment:
Same as above, we are changing the `elements` in the following loop but don't modify the `kinds`. This might not be the expected behavior.

Reply:
In principle yes, but `kinds` are not necessarily related to `elements`. I am a bit hesitant to remove additional information from the `Structure` object that may be needed (e.g. if a specific site is marked regardless of the element it hosts, etc.).

Maybe an additional parameter should be implemented to cover both use cases, something like `remove_kind` or `erase_kind`? That way, at least from the doc-strings the behavior should be clear?

Reply:
Yes, totally agree. I was thinking about something similar, to allow the user to choose. 

Having in mind the use-case that you described, where the kinds are used to label certain sites, it definitely makes sense to keep them.
However, when using the kinds to distinguish magnetic moments, it might be unexpected to find `Co1` and `Co2` kinds in a `Ni` structure, just as an artificial example.